python_modules/libraries/dagster-sdf/dagster_sdf/sdf_information_schema.py

```diff
@@ -139,9 +157,88 @@ def build_sdf_multi_asset_args(
                     )  # Otherwise, use the translator to get the asset key
                     for dep in table_row["depends_on"]
                 }.union(table_id_to_upstream.get(table_row["table_id"], set()))
-
         return deps, outs, internal_asset_deps
 
+    def get_columns(self) -> Dict[str, List[TableColumn]]:
+        columns = self.read_table("columns")[
+            ["table_id", "column_id", "classifiers", "column_name", "datatype", "description"]
+        ]
+        table_columns: Dict[str, List[TableColumn]] = {}
+        for row in columns.rows(named=True):
+            if row["table_id"] not in table_columns:
+                table_columns[row["table_id"]] = []
+            table_columns[row["table_id"]].append(
+                TableColumn(
+                    name=row["column_name"],
+                    type=row["datatype"],
+                    description=row["description"],
+                    constraints=TableColumnConstraints(other=row["classifiers"]),
+                )
+            )
+        return table_columns
+
+    def _extract_code_ref(
+        self, table_row: Dict[str, Any]
+    ) -> Union[CodeReferencesMetadataSet, None]:
+        code_references = None
+        # Check if any of the source locations are .sql files, return the first one
+        loc = (
+            next(
+                (
+                    source_location
+                    for source_location in table_row["source_locations"]
+                    if source_location.endswith(".sql")
+                ),
+                None,
+            )
+            or next(
+                (
+                    source_location
+                    for source_location in table_row["source_locations"]
+                    if source_location.endswith(".sdf.yml")
+                ),
+                None,
+            )
+            or "workspace.sdf.yml"
+        )
+        code_references = CodeReferencesMetadataSet(
+            code_references=CodeReferencesMetadataValue(
+                code_references=[
+                    LocalFileCodeReference(file_path=os.fspath(self.workspace_dir.joinpath(loc)))
+                ]
+            )
+        )
+        return code_references
```

Comment:
Code references is for sure a definitional metadata property. This shouldn't need to rely on run time information.